usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh

```diff
@@ -44,8 +45,18 @@
 
 verify_runnable "global"
 
+function cleanup
+{
+	if is_swap_inuse $swapname ; then
+		log_must $SWAP -d $swapname
```

Comment:
If you change this function to this:

is_swap_inuse $swapname && log_must $SWAP -d $swapname
datasetexists $vol && log_must $ZFS destroy $vol

then you don't need the last two lines of this diff, where the swap gets deleted and the volume destroyed. The cleanup function will handle it automatically.